From 84711d4e605b4ab66a8f53285d34b2c6158683e5 Mon Sep 17 00:00:00 2001
From: Lu Jialin <lujialin4@huawei.com>
Date: Tue, 9 Aug 2022 23:06:50 +0800
Subject: [PATCH] Revert "cgroup: fix compile error when CONFIG_MEMCG = n"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 84711d4e605b4ab66a8f53285d34b2c6158683e5
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4IMAK
CVE: NA

-------------------------------

This reverts commit 58824c83239b31c76f855366c03833deaeb4c982.

Signed-off-by: Lu Jialin <lujialin4@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/vmscan.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 1bfbe1fc67d0..5b5cc00b195b 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -2839,7 +2839,6 @@ static inline bool should_continue_reclaim(struct pglist_data *pgdat,
 
 static bool is_memcg_kswapd_stopped(struct scan_control *sc)
 {
-#ifdef CONFIG_MEMCG
 	struct mem_cgroup *memcg = sc->target_mem_cgroup;
 	bool is_stop = false;
 	unsigned long stop_flag = 0;
@@ -2855,9 +2854,6 @@ static bool is_memcg_kswapd_stopped(struct scan_control *sc)
 	is_stop = page_counter_read(&memcg->memory) < stop_flag;
 
 	return (current_is_kswapd() && is_stop);
-#else
-	return false;
-#endif
 }
 
 static void shrink_node_memcgs(pg_data_t *pgdat, struct scan_control *sc)
-- 
2.33.0

